context.go

```diff
@@ -0,0 +1,37 @@
+package zlog
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/baggage"
+)
+
+// ContextWithValues is a helper for the go.opentelemetry.io/otel/baggage v1
+// API. It takes pairs of strings and adds them to the Context via the baggage
+// package.
+//
+// Any trailing value is silently dropped.
+func ContextWithValues(ctx context.Context, pairs ...string) context.Context {
+	var err error
+	b := baggage.FromContext(ctx)
+	pairs = pairs[:len(pairs)-len(pairs)%2]
+	ms := append(b.Members(), make([]baggage.Member, len(pairs)/2)...)
+	for mi, pi := b.Len(), 0; mi < len(ms); mi, pi = mi+1, pi+2 {
+		k, v := pairs[pi], pairs[pi+1]
+		ms[mi], err = baggage.NewMember(k, v)
```

Comment:
Because `NewMember` might fail, this could result in `nil` members which probably won't be good. You need to either switch to an `append`-style implementation like in my PR, or do `mi--` if `err != nil`, set `ms = ms[:mi]` at the end, and switch the loop condition to look at `pi < len(pairs)` instead. I personally would prefer `append`